### PR TITLE
:recycle: Refactor the tabs component to be reused

### PR DESCRIPTION
**Pull request by foysalit**
This is a simple refactor of the existing tabs components that was just copy pasted across various pages.

### CODE REVIEW

File: components/shell/MobileMenu.tsx [hunk 1]
```diff
@@ -119,9 +119,6 @@ export function MobileMenu({ toggleTheme }: { toggleTheme: () => void }) {
                   <nav className="flex h-full flex-col">
                     <div className="space-y-1">
                       {NAV_ITEMS.map((item) => {
-                        if (item.serviceAccountOnly && !isServiceAccount) {
-                          return
-                        }
                         const Icon = ICONS[item.icon]
```
Comment (devinivy) [inline components/shell/MobileMenu.tsx:122]: Does this end up showing the "Configure" nav item to non-service accounts?  The main issue there is that only the service account has the ability to modify their service record, so it wont have the intended effect if another user does it.

Comment (devinivy) [reply components/shell/MobileMenu.tsx:122]: (There's also similar logic in SidebarNav.tsx)

File: components/common/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}: {
+  currentView: ViewName
+  onSetCurrentView: (v: ViewName) => void
+  views: TabView<ViewName>[]
+}) {
+  const Tab = ({
+    view,
+    label,
+    sublabel,
+  }: {
```
Comment (devinivy) [inline components/common/Tabs.tsx:22]: Defining a component anonymously like this has a quirk, which is that it always fully unmounts and remounts on render.  I'd recommend just pulling this out into its own component outside of `Tabs` render function.

File: components/shell/MobileMenu.tsx (hunk 1 above)
Comment (foysalit) [reply components/shell/MobileMenu.tsx:122]: oh this wasn't meant to be committed. cleaned up.

**Review by devinivy — APPROVED** [review]
Nice!

**Pull request merged**